plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -42,17 +43,16 @@ var BasicsChart = React.createClass({
     bgClasses: React.PropTypes.object.isRequired,
     bgUnits: React.PropTypes.string.isRequired,
     onSelectDay: React.PropTypes.func.isRequired,
+    patient: React.PropTypes.object.isRequired,
     patientData: React.PropTypes.object.isRequired,
     timePrefs: React.PropTypes.object.isRequired,
     updateBasicsData: React.PropTypes.func.isRequired,
+    updateBasicsSettings: React.PropTypes.func.isRequired,
     trackMetric: React.PropTypes.func.isRequired,
   },
   _adjustSectionsBasedOnAvailableData: function(basicsData) {
-    if (_.isEmpty(basicsData.data.reservoirChange.data)) {
-      var siteChangeSection = _.find(basicsData.sections, function(section) {
-        return section.type === 'reservoirChange';
-      });
-      siteChangeSection.active = false;
+    if (basicsData.sections.siteChanges.type !== constants.SECTION_TYPE_UNDECLARED && _.isEmpty(basicsData.data[basicsData.sections.siteChanges.type].data)) {
```

Comment:
Ok, it wasn't clear to me that we don't store the preference of `reservoirChange` for OmniPod, so that situation I constructed locally is definitely fake.

I'm still concerned about this: "Because we have to look back to the time prior to the start of the time period covered by the Basics to calculate the length of the first site change, we put all of the prime & reservoirChange events in the basics data, but if they are all > 3 weeks ago, then the array will have content but the relevant data is still missing."

There's a lot of codepaths in this feature work, and though it does _seem_ like there isn't a path that would result in the kind of mismatch where the site changes data is empty for the basics date range because it's outdated, I think it's possible that there's a combination of circumstances we're not seeing that could result in this. If we could replace the `_.isEmpty` with something a bit more nuanced (like a `_.some` check for date with `time` >= the `basicsData.dateRange[0]`) I think that would be safer.